src/smc-webapp/frame-editors/code-editor/actions.ts

```diff
@@ -81,6 +81,7 @@ export interface CodeEditorState {
   read_only: boolean;
   settings: Map<string, any>; // settings specific to this file (but **not** this user or browser), e.g., spell check language.
   complete: Map<string, any>;
+  derived_file_types: Set<string>;
```

Comment:
this is new: down below is code to check which expected compiled files do really exist. this set will contain nothing, just `"html"`, or `"html", "pdf"`. it's only used for Rmd.